### ISSUE TITLE
Web Service

### ISSUE THREAD

**Issue by laszewsk** [cyberaide/bookmanager-service#2]
Bookmanager Task 2:

 

chose a Web development framework in python or Javascript that allows this to be hosted on a Web page.

Python flask sis super simple and can be learned in 5 minutes from the Web page.

We do not want to do any development in java as this is typically more complex. We want to stay either 100% in javascript/html or a mix out of javascript/html/python

 

We need the following function

 

a) upload a yaml file to the service

b) generate the book from the yaml file

c) e-mail a download link 

d) once a book is created keep it in a repo. This way others could pick up the book

e) develop a cron like behavior where new versions of the book are created on a schedule

 

Next:

deal with user login and 1K books created. E.g. use process pool for parallel creation of multiple books at a ti,

develop a queue using pythons build in features, or add a backend database such as sqlite(easy) or mongodb(complex) 

 

You can use containers to develop this application. It could initially run locally.

 

If you use a javascript framework, electron has been pointed out to be good

**Comment by nchaudh03** [comment]
I have created a Flask App with the core HTML structure. Still, we need to implement Kang's' code, to create a dynamic list of books and chapters. Once that is done need to focus on implementing jQuery  Tree and a button that will push the selected info into a YAML file and eventually service.  CSS is not my priority currently. Please review the core structure and see if that should work.  Branch Naimesh - committed.

**Comment by laszewsk** [comment]
please provide direct urls to the code so we can review easier when you post comments thanks

**Comment by laszewsk** [comment]
eg i can not find it in github, so its not exiting for me ;-)

**Comment by nchaudh03** [comment]
Below is the link to the app.

https://github.com/cyberaide/bookmanager-service/tree/Naimesh/FlaskApp

**Comment by laszewsk** [comment]
the content within 

https://github.com/cyberaide/bookmanager-service/tree/Naimesh/FlaskApp

this should be moved to 

https://github.com/cyberaide/bookmanager-service/tree/Naimesh/cloudmesh/bookmanagerservice/service

then https://github.com/cyberaide/bookmanager-service/tree/Naimesh/FlaskApp

should be deleted

**Comment by nchaudh03** [comment]
I do not see a service folder in bookmanager-service, do I create one and move the entire flask APP folder in, or copy the app.py content into the service.py and then move the template and utils folders into the bookmanager-service folder.

**Comment by laszewsk** [comment]
yes you do create it ;-) you do not move your folder, but all files in that folder and its subfolders

also never name folders with capital letters

eg. all files in FlaskApp/*

**Comment by nchaudh03** [comment]
The app folder has been renamed to service and moved under bookmanager-service directory. 

https://github.com/cyberaide/bookmanager-service/tree/Naimesh/cloudmesh/bookmanagerservice
